File: Byte&Buy.AngularSPA/src/app/features/offers/api-dto/sale-offer-details-response.ts
import { Guid } from "guid-typescript";
import { MoneyDto } from "../../../shared/api-dto/money-dto";
import { CompanySellerResponse } from "./company-seller-response";
import { PrivateSellerResponse } from "./private-seller-response";
import { ImageResponse } from "../../../shared/api-dto/image-response";

export interface SaleOfferDetailsResponse {
    id: Guid;
    quantityAvaliable: number;
    pricePerItem: MoneyDto;
    condition: string;
    category: string;
    description: string;
    title: string;
    seller: CompanySellerResponse | PrivateSellerResponse;
    images: ImageResponse[];
}

File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Public/Offer/SaleOffer/SaleOfferDetailsResponse.cs
using ByteBuy.Core.Domain.Offers.Enums;
using ByteBuy.Core.DTO.Public.Image;
using ByteBuy.Core.DTO.Public.Money;

namespace ByteBuy.Core.DTO.Public.Offer.SaleOffer;

public record SaleOfferDetailsResponse(
    Guid Id,
    int QuantityAvaliable,
    MoneyDto PricePerItem,
    OfferStatus Status,
    string Condition,
    string Category,
    string Description,
    string Title,
    Object Seller,
    IReadOnlyCollection<ImageResponse> Images);